### ISSUE TITLE
salt-ssh: [WARNING ] Failed to spawn the VT: [Errno 6] Device not configured: '/dev/tty'

### ISSUE THREAD

**Issue by nbari** [saltstack/salt#18714]
I am trying to test the salt-ssh (Salt Rosters) but getting this warning and exception:

```
> salt-ssh 'xen1' test.ping --priv /home/nbari/.ssh/id_rsa  
[WARNING ] Failed to spawn the VT: [Errno 6] Device not configured: '/dev/tty'
Process Process-1:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/local/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python2.7/site-packages/salt/client/ssh/__init__.py", line 315, in handle_routine
    stdout, stderr, retcode = single.run()
  File "/usr/local/lib/python2.7/site-packages/salt/client/ssh/__init__.py", line 633, in run
    stdout, stderr, retcode = self.cmd_block()
  File "/usr/local/lib/python2.7/site-packages/salt/client/ssh/__init__.py", line 819, in cmd_block
    stdout, stderr, retcode = self.shell.exec_cmd(cmd_str)
  File "/usr/local/lib/python2.7/site-packages/salt/client/ssh/shell.py", line 284, in exec_cmd
    ret = self._run_cmd(cmd)
  File "/usr/local/lib/python2.7/site-packages/salt/client/ssh/shell.py", line 312, in _run_cmd
    stream_stderr=False)
  File "/usr/local/lib/python2.7/site-packages/salt/utils/vt.py", line 212, in __init__
    'Failed to spawn the VT. Error: {0}'.format(err)
TerminalException: Failed to spawn the VT. Error: [Errno 6] Device not configured: '/dev/tty'
```

My roaster file:

```
> cat /usr/local/etc/salt/roster 
xen1:
    host: 10.15.12.23
    user: nbari
```

The salt master --versions-report is:

```
> salt --versions-report
           Salt: 2014.7.0
         Python: 2.7.8 (default, Dec  4 2014, 09:32:53)
         Jinja2: 2.7.3
       M2Crypto: 0.22
 msgpack-python: 0.4.2
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.11
          ioflo: Not Installed
          PyZMQ: 14.3.1
           RAET: Not Installed
            ZMQ: 4.0.4
           Mako: Not Installed
```

The operating system: FreeBSD 10.1.

Any ideas or posible fix?

**Comment by jfindlay** [comment]
@nbari, are you still having this issue?  Have you tried upgrading to 2014.7.1?

**Comment by nbari** [comment]
haven't tried with 2014.7.1, I am waiting for the FreeBSD port to be updated so that later I could test.

**Comment by nbari** [comment]
Just tested with version **2014.7.2** but still having the same issue

**Comment by 1exx** [comment]
I think it was fixed by #22120
```
# uname -or
FreeBSD 10.2-RELEASE-p18
```
```
# salt-ssh 'testw' test.ping --priv pki/master/ssh/salt-ssh.rsa
testw:
    True
```
roaster file:
```
# cat roster
testw:
    host: 172.16.0.17
    user: Salt
```
salt --version-report is:
```
Salt Version:
           Salt: 2016.3.4
```

**Comment by gtmanfred** [comment]
I am closing this issue, because we are successfully able to use private keys, but because of #22120 and not allocating the tty, we are still unable to use passwords.

Closeing in favor of https://github.com/saltstack/salt/issues/37728
